src/main/java/org/springframework/guice/module/SpringModule.java

```diff
@@ -24,58 +24,95 @@
 import org.springframework.beans.factory.support.AbstractBeanDefinition;
 import org.springframework.beans.factory.support.DefaultListableBeanFactory;
 import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.guice.injector.GuiceAutowireCandidateResolver;
 import org.springframework.util.ClassUtils;
 
+import com.google.inject.AbstractModule;
 import com.google.inject.Binder;
-import com.google.inject.Module;
+import com.google.inject.Injector;
 import com.google.inject.Provider;
 import com.google.inject.ProvisionException;
+import com.google.inject.Stage;
 import com.google.inject.name.Names;
 
 /**
  * @author Dave Syer
  *
  */
-public class SpringModule implements Module {
-
-	private DefaultListableBeanFactory beanFactory;
+public class SpringModule extends AbstractModule {
 
 	private BindingTypeMatcher matcher = new GuiceModuleMetadata();
 
 	private Map<Class<?>, Provider<?>> bound = new HashMap<Class<?>, Provider<?>>();
-
-	public SpringModule(ApplicationContext context) {
-		this((DefaultListableBeanFactory) context.getAutowireCapableBeanFactory());
-	}
-
-	public SpringModule(DefaultListableBeanFactory beanFactory) {
-		this.beanFactory = beanFactory;
-		if (beanFactory.getBeanNamesForType(GuiceModuleMetadata.class).length > 0) {
-			this.matcher = new CompositeTypeMatcher(beanFactory.getBeansOfType(GuiceModuleMetadata.class).values());
-		}
-	}
+	
+    private final Class<?>[] configClasses;
+    private final String[] basePackages;
+    private ApplicationContext context;
+    
+    public SpringModule(Class<?>... configClasses) {
+        this.configClasses = configClasses; 
+        this.basePackages = null;
+    }
+    
+    public SpringModule(String... basePackages) {
+        this.basePackages = basePackages;
+        this.configClasses = null;
+    }
+    
+    public SpringModule(ApplicationContext context) {
```

Comment:
Hmm. I see the problem. But there's no great API to work with config classes before they are a context outside of spring boot. We don't want to depend on Boot here so we'll need to look for something cleverer. Thinking...